com.unity.multiplayer.mlapi/Runtime/Core/RPCQueue/RPCQueueManager.cs

```diff
@@ -0,0 +1,549 @@
+            if (!string.IsNullOrEmpty(channel))
+            {
+                queueHistoryItem.QueueWriter.WriteCharArray(channel.ToCharArray());
+            }
+            else
+            {
+                queueHistoryItem.QueueWriter.WriteByte(0);
+            }
+
+            if (targetNetworkIds != null && targetNetworkIds.Length != 0)
+            {
+                queueHistoryItem.QueueWriter.WriteInt32(targetNetworkIds.Length);
+
+                for (int i = 0; i < targetNetworkIds.Length; i++)
+                {
+                    queueHistoryItem.QueueWriter.WriteUInt64(targetNetworkIds[i]);
+                }
+            }
+            else
+            {
+                queueHistoryItem.QueueWriter.WriteInt32(0);
+            }
+
+            //Mark where we started in the stream to later determine the actual RPC message size (position before writing RPC message vs position after write has completed)
+            queueHistoryItem.MarkCurrentStreamPosition();
+
+            //Write a filler dummy size of 0 to hold this position in order to write to it once the RPC is done writing.
+            queueHistoryItem.QueueWriter.WriteInt64(0);
+
+            //Return the writer to the invoking method.
+            return queueHistoryItem.QueueWriter;
+        }
+
+        /// <summary>
+        /// EndAddQueueItemToOutboundFrame
+        /// Signifies the end of this outbound RPC.
+        /// We store final MSG size and track the total current frame queue size
+        /// </summary>
+        /// <param name="writer">writer that was used</param>
+        public void EndAddQueueItemToOutboundFrame(BitWriter writer)
+        {
+            QueueHistoryFrame queueHistoryItem = GetCurrentQueueHistoryFrame(QueueHistoryFrame.QueueFrameType.Outbound);
+            PooledBitWriter pbWriter = (PooledBitWriter)writer;
+
+            //Sanity check
+            if (pbWriter != queueHistoryItem.QueueWriter)
+            {
+                UnityEngine.Debug.LogError("RPCQueueManager " + QueueHistoryFrame.QueueFrameType.Outbound + " passed writer is not the same as the current PooledBitWrite for the " + QueueHistoryFrame.QueueFrameType.Outbound + "]!");
+            }
+
+            //The total size of the frame is the last known position of the stream
+            queueHistoryItem.TotalSize = (uint)queueHistoryItem.QueueStream.Position;
+
+            long CurrentPosition = queueHistoryItem.QueueStream.Position;
+            ulong BitPosition = queueHistoryItem.QueueStream.BitPosition;
+
+            //////////////////////////////////////////////////////////////
+            //>>>> REPOSITIONING STREAM TO RPC MESSAGE SIZE LOCATION <<<<
+            //////////////////////////////////////////////////////////////
+            queueHistoryItem.QueueStream.Position = queueHistoryItem.GetCurrentMarkedPosition();
+
+            Int64 MSGSize = (Int64)(queueHistoryItem.TotalSize - (queueHistoryItem.GetCurrentMarkedPosition() + 8));
```

Comment:
very minor but I think we should use `long` instead of `Int64` here.

Reply:
Will be added to the RPCQueue revisited.